ivadomed/loader/bids_dataframe.py

```diff
@@ -174,23 +190,30 @@ def add_tsv_metadata(self, df, path_data, layout):
             layout (BIDSLayout): pybids BIDSLayout of the indexed files of the path_data
         """
 
-        # Add participant_id column, and metadata from participants.tsv file if present
+        # Drop columns with all null values before loading TSV metadata
+        # Avoid conflicts with unused columns descriptions from TSV sidecar JSON files
+        df.dropna(axis=1, inplace=True, how='all')
+
+        # Add metadata from 'participants.tsv' file if present
         # Uses pybids function
-        df['participant_id'] = "sub-" + df['subject']
         if layout.get_collections(level='dataset'):
             df_participants = layout.get_collections(level='dataset', merge=True).to_df()
+            df_participants.insert(1, 'participant_id', "sub-" + df_participants['subject'])
             df_participants.drop(['suffix'], axis=1, inplace=True)
             df = pd.merge(df, df_participants, on='subject', suffixes=("_x", None), how='left')
 
-        # Add sample_id column if sample column exists, and add metadata from samples.tsv file if present
+        # Add metadata from 'samples.tsv' file if present
+        # The 'participant_id' column is added only if not already present from the 'participants.tsv' file.
         # TODO: use pybids function after Microscopy-BIDS is integrated in pybids
-        if 'sample' in df:
-            df['sample_id'] = "sample-" + df['sample']
         fname_samples = Path(path_data, "samples.tsv")
         if fname_samples.exists():
             df_samples = pd.read_csv(str(fname_samples), sep='\t')
-            df = pd.merge(df, df_samples, on=['participant_id', 'sample_id'], suffixes=("_x", None),
-                          how='left')
+            df_samples['sample'] = df_samples['sample_id'].str.split("sample-").apply(lambda x: x[1])
+            df_samples['subject'] = df_samples['participant_id'].str.split("sub-").apply(lambda x: x[1])
+            columns = df_samples.columns.tolist()
+            if 'participant_id' in df.columns:
+                columns.remove('participant_id')
+            df = pd.merge(df, df_samples[columns], on=['subject', 'sample'], suffixes=("_x", None), how='left')
```

Comment:
Need to test this more interactively. To be updated post follow-up.

Reply:
@mariehbourget I have stepped through these code as  they are currently implemented, while I was able to follow what it was doing in terms manipulating both `df_participants` and `df_samples` and then modifying them such that they eventually merge into the df. 

Could you maybe explain a bit more about the context and core rationale behind this change? vs the old way of doing merge?

Reply:
I'll refer you to point 1 the first description of the PR [here](https://github.com/ivadomed/ivadomed/pull/1112#issue-1177327116), and try to explain it in other words here as well, sorry I know this is hard to follow and I appreciate your time reviewing it.

I'll take a microscopy example because it contains both subjects (participants) and samples.

In BIDS:
- a "subject" is the `XX` in a sub-XX folder.
- a "participant_id" is the same with the sub- prefix, so `sub-XX`.
- a "sample" is the `YY` in a sample-YY filename part.
- a "sample_id" is the same with the sample- prefix, so `sample-YY`.

You can check the bids_dataframe.csv file to see those 4 columns.

The old way:
1. All the files in the dataset were indexed in the df with 1 row/file --> that create the "subject" and "sample" columns in the main df.
2. The "participant_id" and sample_id" columns were added based on the content of  the "subject" and "sample" columns in the main df.
3. The content of the TSV files were read in `df_participants` and `df_samples`.
4. The content of these df were merged in the main df on "subject" for `participants.tsv` and on "participant_id" and "sample_id" for `samples.tsv.`

Why is it a problem:
At point 2 above: we were assigning the columns "participant_id" and "sample_id" BEFORE reading the TSV file. So there was a strong assumption that all the "participant_id" and "sample_id" were present in the TSV files, which SHOULD be always true but is unfortunately not.

The new way:
1. All the files in the dataset were indexed in the df with 1 row/file --> that create the "subject" and "sample" columns in the main df.
2. Step 2 from above is skipped.
3. The content of the TSV files were read in `df_participants` and `df_samples`.
4. The content of these df are merged in the main df on "subject" for `participants.tsv` and on "subject" and "sample" for `samples.tsv`.

What is the difference?
In the new way, the resulting "participant_id" and "sample_id" columns directly reflects the content of the `participants.tsv` and `samples.stv` files without any assumptions.